Typings are not compatible with `moduleResolution=NodeNext`

If you use the library from the project which `tsconfig.json` has

```json
{
  "compilerOptions": {
    "moduleResolution": "NodeNext",
    "skipLibCheck": false,
    "...": "..."
  }
}
```

And run `tsc --build`

```
node_modules/spacetime/types/index.d.ts:1:38 - error TS2834: Relative import paths need explicit file extensions in ECMAScript imports when '--moduleResolution' is 'node16' or 'nodenext'. Consider adding an extension to the import path.

1 import type { SpacetimeStatic } from './constructors'
                                       ~~~~~~~~~~~~~~~~

node_modules/spacetime/types/index.d.ts:8:84 - error TS2834: Relative import paths need explicit file extensions in ECMAScript imports when '--moduleResolution' is 'node16' or 'nodenext'. Consider adding an extension to the import path.

8 export { SpacetimeConstructor, SpacetimeConstructorOptions, SpacetimeStatic } from './constructors'
                                                                                     ~~~~~~~~~~~~~~~~

node_modules/spacetime/types/index.d.ts:9:47 - error TS2834: Relative import paths need explicit file extensions in ECMAScript imports when '--moduleResolution' is 'node16' or 'nodenext'. Consider adding an extension to the import path.

9 export { Format, I18nOptions, TimeUnit } from './constraints'
                                                ~~~~~~~~~~~~~~~

node_modules/spacetime/types/index.d.ts:10:91 - error TS2834: Relative import paths need explicit file extensions in ECMAScript imports when '--moduleResolution' is 'node16' or 'nodenext'. Consider adding an extension to the import path.

10 export { Spacetime, Diff, ParsableDate, Progress, Since, TimezoneMeta, TimezoneSet } from './types'        
                                                                                             ~~~~~~~~~        


Found 4 errors.
```

--- a/types/constructors.d.ts
+++ b/types/constructors.d.ts
@@ -1,4 +1,4 @@
-import type { Spacetime, ParsableDate, TimezoneSet } from './types'
+import type { Spacetime, ParsableDate, TimezoneSet } from './types.js'
 
 export interface SpacetimeConstructorOptions {
   /** javascript dates use millisecond-epochs, instead of second-epochs, like some other languages. This is a common bug, and by default spacetime warns if you set an epoch within January 1970. to disable set to true */

--- a/types/index.d.ts
+++ b/types/index.d.ts
@@ -1,10 +1,11 @@
-import type { SpacetimeStatic } from './constructors'
+import type { SpacetimeStatic } from './constructors.js'
 
 declare const spacetime: SpacetimeStatic
 
 // We need to use a single default export here so everything lines up with the actual imported object from JS
 export default spacetime
 
-export { SpacetimeConstructor, SpacetimeConstructorOptions, SpacetimeStatic } from './constructors'
-export { Format, I18nOptions, TimeUnit } from './constraints'
-export { Spacetime, Diff, ParsableDate, Progress, Since, TimezoneMeta, TimezoneSet } from './types'
+
+export { SpacetimeConstructor, SpacetimeConstructorOptions, SpacetimeStatic } from './constructors.js'
+export { Format, I18nOptions, TimeUnit } from './constraints.js'
+export { Spacetime, Diff, ParsableDate, Progress, Since, TimezoneMeta, TimezoneSet } from './types.js'

--- a/types/types.d.ts
+++ b/types/types.d.ts
@@ -1,4 +1,4 @@
-import type { TimeUnit, Format, I18nOptions } from './constraints'
+import type { TimeUnit, Format, I18nOptions } from './constraints.js'
 
 /** a date/timezone object */
 export interface Spacetime {

